tests/kernels/test_cutlass.py

```diff
@@ -0,0 +1,196 @@
+"""Tests for cutlass kernels
+
+Run `pytest tests/kernels/test_cutlass.py`.
+"""
+from typing import Type
+
+import pytest
+import torch
+
+from vllm import _custom_ops as ops
+
+CUDA_DEVICES = [
+    f"cuda:{i}" for i in range(1 if torch.cuda.device_count() == 1 else 2)
+]
+
+capability = torch.cuda.get_device_capability()
+capability = capability[0] * 10 + capability[1]
+
+
+def to_fp8(tensor: torch.tensor):
+    # Assuming input tensor is float32
+    # Scale tensor to range of FP8 E4M3
+    # by clamping exponent and truncating mantissa
```

Comment:
Hehe I guess it works but the seems slightly odd -- why not go with a more traditional strategy like in https://github.com/vllm-project/vllm/blob/361c461a128a5df2faefeb70ffa98e61e4feda55/vllm/model_executor/layers/quantization/fp8.py#L261 and just clamp between min and max?

I guess the way it is right now will generate more diverse test data, but then in practice people will use the clamping between min and max (and that's also what you do for int8). You can use the same `* 5` that you use in cutlass_int8_gemm_helper in cutlass_fp8_gemm_helper too and that would probably be diverse enough without this odd conversion technique :)

Reply:
yes much better :sweat_smile: